icetransport.go

```diff
@@ -200,28 +204,31 @@ func (t *ICETransport) GracefulStop() error {
 
 func (t *ICETransport) stop(shouldGracefullyClose bool) error {
 	t.lock.Lock()
-	defer t.lock.Unlock()
-
 	t.setState(ICETransportStateClosed)
 
 	if t.ctxCancel != nil {
 		t.ctxCancel()
 	}
 
+	// mux and gatherer can only be set when ICETransport.State != Closed.
+	mux := t.mux
+	gatherer := t.gatherer
+	t.lock.Unlock()
+
 	if t.mux != nil {
 		var closeErrs []error
-		if shouldGracefullyClose && t.gatherer != nil {
```

Comment:
I looked at these closes and they should be safe to call concurrently. If they aren't, we'll see some other problems crop up quickly. But I haven't found those to happen in any concurrent PC close test runs.